modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1588,22 +1588,28 @@ def shuffle_partitions(
             partition.force_materialization().list_of_block_partitions[0]
             for partition in cls.row_partitions(partitions)
         ]
-        # Gather together all of the sub-partitions
-        split_row_partitions = np.array(
-            [
-                partition.split(
-                    shuffle_functions.split_function, len(pivots) + 1, pivots
-                )
-                for partition in row_partitions
-            ]
-        ).T
-        # We need to convert every partition that came from the splits into a full-axis column partition.
-        new_partitions = [
-            [
-                cls._column_partitions_class(row_partition, full_axis=False).apply(
-                    final_shuffle_func
-                )
+        if len(pivots):
```

Comment:
2nd safeguard:
If we have no pivots don't go to the splitting logic, but apply a simple row-axis function instead